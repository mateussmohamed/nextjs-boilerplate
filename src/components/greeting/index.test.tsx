import { render, screen } from '@testing-library/react'

import Greeting from '.'

describe('<Greeting>', () => {
  it('should render the greeting', () => {
    render(<Greeting />)

    expect(
      screen.getByRole('heading', { name: /react boilerplate/i })
    ).toBeInTheDocument()
  })

  it('shoulder render correctly', () => {
    const container = render(<Greeting />)

    expect(container).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Greeting />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
  })
})
