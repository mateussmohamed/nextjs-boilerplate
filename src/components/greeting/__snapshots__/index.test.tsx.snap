// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Greeting> shoulder render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

.c4 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

<body>
    <div>
      <main
        class="c0"
      >
        <img
          alt="Imagem de um átomo e React Avançado escrito ao lado."
          class="c1"
          src="/img/logo.svg"
        />
        <h1
          class="c2"
        >
          React Boilerplate
        </h1>
        <h2
          class="c3"
        >
          TypeScript, ReactJS, NextJS e Styled Components
        </h2>
        <img
          alt="Um desenvolvedor de frente para uma tela com código."
          class="c4"
          src="/img/hero-illustration.svg"
        />
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="styles__Wrapper-lstnt1-0 cedmqx"
    >
      <img
        alt="Imagem de um átomo e React Avançado escrito ao lado."
        class="styles__Logo-lstnt1-1 isYCES"
        src="/img/logo.svg"
      />
      <h1
        class="styles__Title-lstnt1-2 dhQoZo"
      >
        React Boilerplate
      </h1>
      <h2
        class="styles__Description-lstnt1-3 gfEPvP"
      >
        TypeScript, ReactJS, NextJS e Styled Components
      </h2>
      <img
        alt="Um desenvolvedor de frente para uma tela com código."
        class="styles__Illustration-lstnt1-4 kFfUJW"
        src="/img/hero-illustration.svg"
      />
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`
